share/lib/python/neuron/rxd/species.py

```diff
@@ -781,38 +784,71 @@ def _transfer_to_legacy(self):
                 # TODO: vectorize this, don't recompute denominator unless a structure change event happened
                 ptr[0] = sum(nodes[node].concentration * nodes[node].volume for node in all_nodes_in_seg) / sum(nodes[node].volume for node in all_nodes_in_seg)
 
-    def _parse_diffusion(self,d):
+    def _parse_diffusion(self, d):
+        def get_signature(fun):
+            import sys
+            import inspect
+            if sys.version_info.major > 2:
+                sig = inspect.signature(fun)
+                for param in sig.parameters.values():
+                    if (param.kind == param.VAR_POSITIONAL or
+                        param.kind == param.VAR_KEYWORD):
+                        raise RxDException("Intracellular diffusion coefficient function may not include *args or *kwargs")
+                return(len(sig.parameters))
+            else:
+                sig = inspect.getargspec(fun)
+                if sig.args != None or sig.keywords != None:
+                    raise RxDException("Intracellular diffusion coefficient function may not include *args or *kwargs")
+                return len(sig.args)
         dc = None
         dgrid = None
         if callable(d) or (hasattr(d, '__len__') and callable(d[0])):
-            dgrid = numpy.ndarray((3, self._nodes_length), dtype=float,
+            dgrid = numpy.ndarray((3, self._nodes_length), dtype=numpy.float64,
                                   order='C')
+            from inspect import signature
```

Comment:
doesn't this fail in Python 2? For all the reasons that you have the above code block with inspect.getargspec